### ISSUE TITLE
After first building, add new codes to Program.cs of .net core project and build again does not work well.

### ISSUE THREAD

**Issue by Xiu-Li** [dotnet/project-system#3050]
- Create a new .net core console app project, first build, run and debug succeeded.
- The Output window of Build shows as below:
`========== Build: 0 succeeded, 0 failed, 1 up-to-date, 0 skipped ==========`
- Then edit Program.cs file of the project, add some new codes "`Console.WriteLine("Hi Xiu!");`" and save.
- Build the project again, the Output window of Build still shows " `========== Build: 0 succeeded, 0 failed, 1 up-to-date, 0 skipped ========== `".
![image](https://user-images.githubusercontent.com/24784908/33932122-11d4f816-e02d-11e7-897e-092f37de37ea.png)
- Run the project again, the new codes added can't be outputted.
- Put a breakpoint in the code. F5, the breakpoint is not hit.
![image](https://user-images.githubusercontent.com/24784908/33932181-37a63d20-e02d-11e7-9b4f-4f77d2fce0b5.png)

VS Build: **D15.6/27212.00 (15.6.0 Preview 2.0)**
dotnet --info:
> .NET Command Line Tools (2.1.4-preview-007237)
> 
> Product Information:
>  Version:            2.1.4-preview-007237
>  Commit SHA-1 hash:  9c5843d071
> 
> Runtime Environment:
>  OS Name:     Windows
>  OS Version:  10.0.10586
>  OS Platform: Windows
>  RID:         win10-x64
>  Base Path:   C:\Program Files\dotnet\sdk\2.1.4-preview-007237\
> 
> Microsoft .NET Core Shared Framework Host
> 
>   Version  : 2.0.4
>   Build    : 7f262f453d8c8479b9af91d34c013b3aa05bc1ff
>

**Comment by panopticoncentral** [comment]
@davkean If I turn on up to date logging, it looks like we are getting nothing from the output group dataflow, so there's no output to compare against. The code hasn't changed, are there any known issues in the dataflow?

**Comment by davkean** [comment]
Yes the dataflow block has changed - will forward you the email.

**Comment by panopticoncentral** [comment]
Actually, this issue is not fixed by the PR. The issue seems to be that CPS sometimes reports output groups, sometimes doesn't. That made it seem like this fixed the problem but it really didn't.
